Scripts/RemediationScripts/Remediate-ConfigureAADAuthExtOnLinuxVMSS.ps1

```diff
@@ -0,0 +1,1109 @@
+                                  @{Expression={$_.OSType};Label="OSType";Width=30;Alignment="left"},
+                                  @{Expression={$_.OrchestrationMode};Label="OrchestrationMode";Width=30;Alignment="left"},
+                                  @{Expression={$_.isExtPresent};Label="isExtensionPresent";Width=30;Alignment="left"},
+                                  @{Expression={$_.isExtInstalledPostRemediation};Label="isExtInstalledPostRemediation";Width=30;Alignment="left"}
+       
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "Remediation Summary: " -ForegroundColor $([Constants]::MessageType.Info)
+
+        if ($($VMSSRemediated | Measure-Object).Count -gt 0)
+        {
+            Write-Host "AAD Extension have been installed on following Virtual Machine Scale Set(s) in the subscription:" -ForegroundColor $([Constants]::MessageType.Update)
+           
+            $VMSSRemediated | Format-Table -Property $colsPropertyRemediated -Wrap
+
+            # Write this to a file.
+            $VMSSRemediatedFile = "$($backupFolderPath)\RemediatedVMSS.csv"
+            $VMSSRemediated | Export-CSV -Path $VMSSRemediatedFile -NoTypeInformation
+
+            Write-Host "This information has been saved to" -NoNewline
+            Write-Host " [$($VMSSRemediatedFile)]" -ForegroundColor $([Constants]::MessageType.Update) 
+            Write-Host "Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Info)
+        }
+
+        if ($($VMSSSkipped | Measure-Object).Count -gt 0)
+        {
+            Write-Host "Error installing AAD Extension on the following Virtual Machine Scale Set(s) in the subscription: " -ForegroundColor $([Constants]::MessageType.Error)
+            $VMSSSkipped | Format-Table -Property $colsProperty -Wrap
+            # Write this to a file.
+            $VMSSSkippedFile = "$($backupFolderPath)\SkippedVMSS.csv"
+            $VMSSSkipped | Export-CSV -Path $VMSSSkippedFile -NoTypeInformation
+            Write-Host "This information has been saved to"  -NoNewline
+            Write-Host " [$($VMSSSkippedFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+        }
+    }
+    else
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4]  Remediating non compliant Virtual Machines Scale Set(s)..."
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "Skipped as -DryRun switch is provided." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::DoubleDashLine)
+
+        Write-Host "Next steps:" -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host "Run the same command with -FilePath $($backupFile) and without -DryRun to configure AAD Extension on Virtual Machine Scale Set(s) listed in the file."
+    }
+}
+
+function Validate-AADAuthExtensionforVMSS
```

Comment:
Why Validate?

Reply:
Validate checks for provisioning state after installation of extensions.